tests/test_tfparse.py

```diff
@@ -121,4 +121,77 @@ def test_parse_notify_slack(tmp_path):
     assert "notify_slack_qa" in module
     assert "notify_slack_saas" in module
     assert "lambda" in module
-    assert len(module["lambda"]) == 2
+
+    assert len(module["lambda"]) == 36
```

Comment:
This is wrong. Based on this terraform (notify_slack), I would expect lambda to have two entires, one for each underlying lambda being produced by the provider. See the output below.

```
	"module": {
		"lambda": [
			{
				"__tfmeta": {
					"filename": "terraform-aws-modules/notify-slack/aws/main.tf",
					"line_end": 135,
					"line_start": 71
				},
				"allowed_triggers": {
					"AllowExecutionFromSNS": {
						"principal": "sns.amazonaws.com",
						"source_arn": "8166a20f-889f-4487-9c37-07b2d860721c"
					}
				},
				"attach_cloudwatch_logs_policy": false,
				"attach_network_policy": false,
				"attach_policy_json": true,
				"create": true,
				"create_role": true,
				"depends_on": [
					[
						{
							"arn": "3d28c422-b539-49da-8d74-5bd933e56f79",
							"count": 1,
							"id": "3d28c422-b539-49da-8d74-5bd933e56f79",
							"kms_key_id": null,
							"name": "/aws/lambda/notify_slack_qa",
							"retention_in_days": 0,
							"tags": null
						}
					]
				],
				"description": null,
				"environment_variables": {
					"LOG_EVENTS": "False",
					"SLACK_CHANNEL": "feed-ops-qa",
					"SLACK_EMOJI": ":aws:",
					"SLACK_USERNAME": null,
					"SLACK_WEBHOOK_URL": "https://localhost"
				},
				"ephemeral_storage_size": 512,
				"function_name": "notify_slack_qa",
				"handler": "notify_slack.lambda_handler",
				"id": "3e9f9ac3-eeba-4c40-88a0-5097425f3ded",
				"kms_key_arn": "",
				"lambda_role": "",
				"policy_json": "",
				"policy_path": null,
				"publish": true,
				"recreate_missing_package": true,
				"reserved_concurrent_executions": -1,
				"role_name": "-notify_slack_qa",
				"role_path": null,
				"role_permissions_boundary": null,
				"role_tags": {},
				"runtime": "python3.8",
				"s3_bucket": null,
				"source": "terraform-aws-modules/lambda/aws",
				"source_path": "./functions/notify_slack.py",
				"store_on_s3": false,
				"tags": null,
				"timeout": 30,
				"use_existing_cloudwatch_log_group": true,
				"version": "3.2.0",
				"vpc_security_group_ids": null,
				"vpc_subnet_ids": null
			},
			{
				"__tfmeta": {
					"filename": "terraform-aws-modules/notify-slack/aws/main.tf",
					"line_end": 137,
					"line_start": 73
				},
				"allowed_triggers": {
					"AllowExecutionFromSNS": {
						"principal": "sns.amazonaws.com",
						"source_arn": "d0b8b0ce-168a-4b6b-9273-be03a3aab449"
					}
				},
				"attach_cloudwatch_logs_policy": false,
				"attach_network_policy": false,
				"attach_policy_json": true,
				"create": true,
				"create_role": true,
				"depends_on": [
					[
						{
							"arn": "686ece7b-2f5d-4882-a927-54c4b12d79e6",
							"count": 1,
							"id": "686ece7b-2f5d-4882-a927-54c4b12d79e6",
							"kms_key_id": null,
							"name": "/aws/lambda/notify_slack_saas",
							"retention_in_days": 0,
							"tags": null
						}
					]
				],
				"description": null,
				"environment_variables": {
					"LOG_EVENTS": "False",
					"SLACK_CHANNEL": "feed-ops-saas",
					"SLACK_EMOJI": ":aws:",
					"SLACK_USERNAME": null,
					"SLACK_WEBHOOK_URL": "https://localhost"
				},
				"ephemeral_storage_size": 512,
				"function_name": "notify_slack_saas",
				"handler": "notify_slack.lambda_handler",
				"id": "5c8629fd-84cc-4c96-85bc-b2e104f04049",
				"kms_key_arn": "",
				"lambda_role": "",
				"policy_json": "",
				"policy_path": null,
				"publish": true,
				"recreate_missing_package": true,
				"reserved_concurrent_executions": -1,
				"role_name": "-notify_slack_saas",
				"role_path": null,
				"role_permissions_boundary": null,
				"role_tags": {},
				"runtime": "python3.8",
				"s3_bucket": null,
				"source": "terraform-aws-modules/lambda/aws",
				"source_path": "./functions/notify_slack.py",
				"store_on_s3": false,
				"tags": null,
				"timeout": 30,
				"use_existing_cloudwatch_log_group": true,
				"version": "3.2.0",
				"vpc_security_group_ids": null,
				"vpc_subnet_ids": null
			}
		],
```

Your branch produces

```
	"module": {
		"lambda": {
			"__tfmeta": {
				"filename": "terraform-aws-modules/notify-slack/aws/main.tf",
				"line_end": 135,
				"line_start": 71
			},
			"allowed_triggers": {
				"AllowExecutionFromSNS": {
					"principal": "sns.amazonaws.com",
					"source_arn": "71f5b1eb-80ff-43b8-ba43-ed49449da3ca"
				}
			},
			"attach_cloudwatch_logs_policy": false,
			"attach_network_policy": false,
			"attach_policy_json": true,
			"create": true,
			"create_role": true,
			"depends_on": [
				[
					{
						"arn": "df8c2458-a03d-4c6e-a4ed-f0db004abfd7",
						"count": 1,
						"id": "df8c2458-a03d-4c6e-a4ed-f0db004abfd7",
						"kms_key_id": null,
						"name": "/aws/lambda/notify_slack_qa",
						"retention_in_days": 0,
						"tags": null
					}
				]
			],
			"description": null,
			"environment_variables": {
				"LOG_EVENTS": "False",
				"SLACK_CHANNEL": "feed-ops-qa",
				"SLACK_EMOJI": ":aws:",
				"SLACK_USERNAME": "slack-user-2",
				"SLACK_WEBHOOK_URL": "https://localhost"
			},
			"ephemeral_storage_size": 512,
			"function_name": "notify_slack_qa",
			"handler": "notify_slack.lambda_handler",
			"id": "308115f0-b91f-47b8-ae6f-5e04c76e13b1",
			"kms_key_arn": "",
			"lambda_role": "",
			"policy_json": "",
			"policy_path": null,
			"publish": true,
			"recreate_missing_package": true,
			"reserved_concurrent_executions": -1,
			"role_name": "-notify_slack_qa",
			"role_path": null,
			"role_permissions_boundary": null,
			"role_tags": {},
			"runtime": "python3.8",
			"s3_bucket": null,
			"source": "terraform-aws-modules/lambda/aws",
			"source_path": "./functions/notify_slack.py",
			"store_on_s3": false,
			"tags": null,
			"timeout": 30,
			"use_existing_cloudwatch_log_group": true,
			"version": "3.2.0",
			"vpc_security_group_ids": null,
			"vpc_subnet_ids": null
		},
		"notify_slack_qa": {
			"__tfmeta": {
				"filename": "main.tf",
				"line_end": 41,
				"line_start": 22
			},
			"iam_role_name_prefix": null,
			"id": "7c85520d-e589-44a3-be10-151b33d7fc07",
			"lambda_function_name": "notify_slack_qa",
			"slack_channel": "feed-ops-qa",
			"slack_username": "slack-user-2",
			"slack_webhook_url": "https://localhost",
			"sns_topic_kms_key_id": null,
			"sns_topic_name": "slack-alert-qa",
			"source": "terraform-aws-modules/notify-slack/aws",
			"tags": null,
			"version": "~\u003e 5.1.0"
		},
		"notify_slack_saas": {
			"__tfmeta": {
				"filename": "main.tf",
				"line_end": 20,
				"line_start": 1
			},
			"iam_role_name_prefix": null,
			"id": "66018b0a-2ca4-4eec-af94-02af500dbddd",
			"lambda_function_name": "notify_slack_saas",
			"slack_channel": "feed-ops-saas",
			"slack_username": "slack-user-1",
			"slack_webhook_url": "https://localhost",
			"sns_topic_kms_key_id": null,
			"sns_topic_name": "slack-alert-saas",
			"source": "terraform-aws-modules/notify-slack/aws",
			"tags": null,
			"version": "~\u003e 5.3.0"
		}
	},
```
It is missing the context for the provider created lambdas.

Reply:
Ah, yeah, I saw this earlier, been working on it. This is a fun problem, no? :)